vyper/parser/context.py

```diff
@@ -71,25 +72,35 @@ def __init__(self, vars, global_ctx, sigs=None, forvars=None, return_type=None,
         # store global context
         self.global_ctx = global_ctx
 
-    def set_in_assignment(self, state: bool):
-        self.in_assignment = state
+    def is_constant(self):
+        return self.constancy == Constancy.Constant or self.in_assertion
```

Comment:
For enums it is typically better to do `self.constancy is Constancy.Constant` since enums use singletons.